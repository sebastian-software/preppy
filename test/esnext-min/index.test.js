/* global __dirname */
import { readFile } from "fs"
import { resolve } from "path"

import pify from "pify"
import rimraf from "rimraf"

import { preppy } from "../util"

const lazyRead = pify(readFile)
const lazyDelete = pify(rimraf)


test("Publish Test File via Babel", async () => {
  await lazyDelete(resolve(__dirname, "./dist"))
  await preppy({
    cwd: __dirname
  })

  expect(await lazyRead(resolve(__dirname, "dist/index.cjs.min.js"), "utf8")).toMatchSnapshot(
    "cjs"
  )
  expect(await lazyRead(resolve(__dirname, "dist/index.esm.min.js"), "utf8")).toMatchSnapshot(
    "esm"
  )
  expect(await lazyRead(resolve(__dirname, "dist/index.umd.min.js"), "utf8")).toMatchSnapshot(
    "umd"
  )
})
