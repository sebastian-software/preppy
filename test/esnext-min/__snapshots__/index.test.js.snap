// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Publish Test File via Babel: cjs 1`] = `
"/*! esnext-min v1.0.0 by Sebastian Werner <s.werner@sebastian-software.de> */
\\"use strict\\"
function e(e){return e&&\\"object\\"==typeof e&&\\"default\\"in e?e.default:e}require(\\"core-js/modules/es.promise.js\\")
var o=e(require(\\"lodash/camelCase\\"))
console.log(\\"CherryPick Import Lodash:\\",\\"helloWorld\\"===o(\\"hello world\\")),console.log(\\"Package\\",\\"esnext-min\\"),console.log(\\"Target\\",\\"lib\\"),console.log(\\"ES2016 Enabled:\\",!0),new Promise(((e,o)=>{e(\\"resolved\\")})).then((e=>{console.log(\\"Promise:\\",e)}))
console.log(\\"Constant:\\",123)
var l=[1,2,3]
console.log(\\"Supports Array.includes?:\\",l.includes&&l.includes(2))
var s=new Set(l)
console.log(\\"Supports Set:\\",s.add(4)),function(e=!0){console.log(\\"Supports default parameters:\\",e)}()
{let e=\\"inner\\"
console.log(\\"X Value from inner scope:\\",e)}console.log(\\"X Value from outer scope:\\",\\"outer\\")
var r=Object.assign({third:3},{first:1,second:2})
console.log(\\"Destructed:\\",r)
class n{constructor(){console.log(\\"Called constructor\\"),this.helper(),this.instanceProperty=3}helper(){console.log(\\"Called helper\\")}}n()
console.log(\\"Initialized class:\\",new n),async function(){await PromiseHelper()}()
//# sourceMappingURL=index.cjs.min.js.map
"
`;

exports[`Publish Test File via Babel: esm 1`] = `
"/*! esnext-min v1.0.0 by Sebastian Werner <s.werner@sebastian-software.de> */
import\\"core-js/modules/es.promise.js\\"
import o from\\"lodash/camelCase\\"
console.log(\\"CherryPick Import Lodash:\\",\\"helloWorld\\"===o(\\"hello world\\")),console.log(\\"Package\\",\\"esnext-min\\"),console.log(\\"Target\\",\\"lib\\"),console.log(\\"ES2016 Enabled:\\",!0),new Promise(((o,e)=>{o(\\"resolved\\")})).then((o=>{console.log(\\"Promise:\\",o)}))
console.log(\\"Constant:\\",123)
var e=[1,2,3]
console.log(\\"Supports Array.includes?:\\",e.includes&&e.includes(2))
var l=new Set(e)
console.log(\\"Supports Set:\\",l.add(4)),function(o=!0){console.log(\\"Supports default parameters:\\",o)}()
{let o=\\"inner\\"
console.log(\\"X Value from inner scope:\\",o)}console.log(\\"X Value from outer scope:\\",\\"outer\\")
var s=Object.assign({third:3},{first:1,second:2})
console.log(\\"Destructed:\\",s)
class r{constructor(){console.log(\\"Called constructor\\"),this.helper(),this.instanceProperty=3}helper(){console.log(\\"Called helper\\")}}r()
console.log(\\"Initialized class:\\",new r),async function(){await PromiseHelper()}()
//# sourceMappingURL=index.esm.min.js.map
"
`;

exports[`Publish Test File via Babel: umd 1`] = `
"/*! esnext-min v1.0.0 by Sebastian Werner <s.werner@sebastian-software.de> */
!function(e,o){\\"object\\"==typeof exports&&\\"undefined\\"!=typeof module?o(require(\\"core-js/modules/es.promise.js\\"),require(\\"lodash/camelCase\\")):\\"function\\"==typeof define&&define.amd?define([\\"core-js/modules/es.promise.js\\",\\"lodash/camelCase\\"],o):o(null,(e=\\"undefined\\"!=typeof globalThis?globalThis:e||self)._camelCase)}(this,(function(e,o){\\"use strict\\"
function l(e){return e&&\\"object\\"==typeof e&&\\"default\\"in e?e.default:e}var s=l(o)
console.log(\\"CherryPick Import Lodash:\\",\\"helloWorld\\"===s(\\"hello world\\")),console.log(\\"Package\\",\\"esnext-min\\"),console.log(\\"Target\\",\\"lib\\"),console.log(\\"ES2016 Enabled:\\",!0),new Promise(((e,o)=>{e(\\"resolved\\")})).then((e=>{console.log(\\"Promise:\\",e)}))
console.log(\\"Constant:\\",123)
var n=[1,2,3]
console.log(\\"Supports Array.includes?:\\",n.includes&&n.includes(2))
var r=new Set(n)
console.log(\\"Supports Set:\\",r.add(4)),function(e=!0){console.log(\\"Supports default parameters:\\",e)}()
{let e=\\"inner\\"
console.log(\\"X Value from inner scope:\\",e)}console.log(\\"X Value from outer scope:\\",\\"outer\\")
var t=Object.assign({third:3},{first:1,second:2})
console.log(\\"Destructed:\\",t)
class c{constructor(){console.log(\\"Called constructor\\"),this.helper(),this.instanceProperty=3}helper(){console.log(\\"Called helper\\")}}c()
console.log(\\"Initialized class:\\",new c),async function(){await PromiseHelper()}()}))
//# sourceMappingURL=index.umd.min.js.map
"
`;
